docs/articles/rag-application-communication-system.md

```diff
@@ -0,0 +1,212 @@
+
+* Automated re-routing of user queries. A major issue of more generalist chat systems is deciding 1) whether or not to activate the retrieval of external resources, 2) which collection of resources to prioritize (maybe not all are needed; reducing volume can relieve stress to the system), and 3) when a new round of retrieval becomes necessary (e.g., when the topic of a conversation has changed).
+* Query expansion. LLM and embedding models, because they're trained primarily on complete sentences, struggle with unstructured queries. You may require further contextualization by adding elements (e.g., past queries).
+* Text segmentation (rather than just “chunking”). Proper segmentation has proven to be one of the most powerful performance boosts. But as user traffic increases, and more document coverage is needed, iteratively expanding your RAG system (typically by parsing different sets of resources with existing ones) can become prohibitively costly. Lately, I’ve been experimenting a lot with agnostic encoder models for token-classification tasks to reconstruct text parts (including titles, bibliography) from raw text comprehension, regardless of the original format.
+
+For automated query rerouting, query expansion, and text segmentation, using an LLM is overkill and results in performance lag and degradation. Once properly re-trained, encoder models yield comparable results, are very fast, not expensive to host, and better fitted to the overall purpose of classification. In addition, encoders don't require structured generation, regex, or data constraints - the data design is already built-in.
+
+### 3.3 An alternative to generation: synthetic data curation
+
+In the months to come, I anticipate a lot of research on pre-generation of system input and output - basically, swapping models in production with optimized retrieval of synthetic data. This was the approach retained by Klarna for their integration of ChatGPT. It yields obvious advantages in terms of robustness and security (prompt injection is simply impossible).
+
+As RAG systems scale, pre-generation (solving and storing query answers beforehand) provides significant economies of scale, once you have a general grasp of what queries or topics users ask frequently. Most queries follow a 80-20 rule (80% of questions are about 20% of topics); once a question is solved and stored, there's no need to regenerate it, saving computational resources.
+
+It's also possible to expand the original dataset to include the types of queries your users are likely to write. And you are accidentally recreating with… an instruction dataset, with a couple of queries and answers. It may be high time to consider fine-tuning.
```

Comment:
1. "accidentally recreating... with an instruction dataset" - why "accidentally"? 
2. refers to the pre-generation you were just discussing?
3. What exactly is the rationale here for fine-tuning?

Reply:
Oh here my key point is a dataset with pre-generated queries is already an institution dataset for fine-tuning.

"And at this point, a RAG dataset starts to look like an instruction dataset, structured around a couple of answer and (generated) query. It also becomes a resource for fine-tuning and reconfiguring the model behavior."